sound/soc/sof/sof-audio.h

```diff
@@ -331,6 +331,7 @@ struct snd_sof_widget {
 	int use_count; /* use_count will be protected by the PCM mutex held by the core */
 	int core;
 	int id;
+	int instance_id; /* dynamically set when the widget gets set up in the FW */
```

Comment:
instance_id for widgets will be assigned when they are
set up in the DSP and freed when the widgets are freed.

did you mean 'reset when the widgets are freed? Assign/free is not a very obvious set of dual functions.